scripts/globals/beastmentreasure.lua

```diff
@@ -0,0 +1,378 @@
+    and once at the start of each new game day. Since Peddlestox is disabled in the db by default, we
+    only need to enable her on the appropriate day and disable her on the following day. ]]--
+    local zd = zoneData[zoneid]
+    local peddlestox = GetNPCByID(zd.peddlestox.id)
+
+    if VanadielDayElement() == zd.peddlestox.day then
+        peddlestox:setStatus(dsp.status.NORMAL)
+    elseif peddlestox:getStatus() == dsp.status.NORMAL then
+        --[[
+        TODO: Peddlestox should play a teleport animation and then fade out. On retail this is an 0x03A
+            packet with animation id 122. Currently 0x3A is not implemented in DSP. Retail capture of
+            the correct animation packet is as follows:
+             |  0  1  2  3   4  5  6  7   8  9  A  B   C  D  E  F    | 0123456789ABCDEF
+        -----+----------------------------------------------------  -+------------------
+           0 | 3A 0A B5 00  67 B2 07 01  67 B2 07 01  67 02 67 02    | :...g...g...g.g.
+          10 | 7A 00 00 BF                                           | z...
+            An incomplete but similar 0x38 animation can be achieved with `entityAnimationPacket("shwh")`
+        ]]--
+        peddlestox:setStatus(dsp.status.DISAPPEAR)
+    end
+end
+
+dsp.beastmentreasure.handleQmOnTrigger = function(player,npc,buriedtext,nothingtext)
+    local digsite = getAssignedDigSite(player, zoneData[player:getZoneID()])
+    local qmid = npc:getID()
+
+    if digsite == nil or digsite.id ~= qmid then
+        player:messageSpecial(nothingtext)
+    elseif digsite.id == qmid then
+        -- "It looks like something is buried here. If you had a <pickaxe> you could dig it up."
+        player:messageSpecial(buriedtext, 605)
+    end
+end
+
+dsp.beastmentreasure.handleQmOnTrade = function(player,npc,trade)
+    local zoneid = player:getZoneID()
+    local zd = zoneData[zoneid]
+    local digsite = getAssignedDigSite(player, zd)
+
+    if npcUtil.tradeHasExactly(trade, 605)
+        and player:getVar(zd.statusvar) == QUEST_COMPLETED
+        and npc:getID() == digsite.id then
+            --[[ Event 105 needs args to spawn and animate a treasure chest
+                 Example args from retail capture: 105 123 450762 1745 201805 7 723 490292 4095
+                 An arg in the 5th parameter will spawn and animate a chest somewhere in the zone
+                    based on some internal list in the game files. For example, an arg of 7, as
+                    in the retail capture above, will spawn and animate a chest at Yuhtunga Jungle
+                    qm10. Chests will spawn at a static location regardless of whether or not
+                    coordinates are passed in, and will always be rotated to face the player.
+            ]]--
+            player:startEvent(105, zoneid, 0, 0, 0, digsite.chest)
+    end
+end
+
+dsp.beastmentreasure.handleQmOnEventFinish = function(player,csid)
```

Comment:
You're going to want to check if the person has inventory space before confirming the trade and resetting the quest.

Rewards could potentially be lost otherwise.

Reply:
They actually go to the treasure pool in the case of a full inventory, which is the correct behavior here.